### DISCUSSION TITLE
Initial release of Wax

### DISCUSSION THREAD

**Discussion by jeffbarish** [jeffbarish/wax#3] [Announcements]
Initial release of Wax.

<hr /><em>This discussion was created from the release <a href='https://github.com/jeffbarish/wax/releases/tag/'>Initial release of Wax</a>.</em>

**Comment by jrayner00** [comment]
Jeff, this is exceptional work! I am accruing the required materials to attempt this for my Dad's birthday. This is a total aside, but would it be possible to allow for iso ripping of DVD's? As a music lover, I know he'd love to backup his entire collection (2000+ CDs), but he also has many rare DVD's he'd like to dump, and even if achieved in the background this functionality would be great for him.

Appreciate any advice you can proffer!

Many thanks again for this amazing work,
JR

**Comment by jeffbarish** [comment]
Gee, thanks! I have been working on this project for a long time, so plaudits are gratifying. How did you hear about Wax?

As to your aside, I do not know the answer, but it seems unlikely to me because of the DRM issue. If GStreamer supports DVD ripping, then I suppose that it would be possible to add that functionality. I did a quick search that revealed no suggestions that GStreamer does provide that support. I am also wondering how well DVDs would integrate with a music collection. Having a mix of music and video in the play queue seems weird to me. I have no idea how metadata extraction would work. Obviously, Musicbrainz would not be a useful source. My initial thought is that DVD ripping is a different problem that properly requires a different program.

Let me know how your project goes.

**Comment by mikel963** [comment]
This looks interesting and I am trying to install on WSL but having a few issues 
Wax runs but wax-config does not - with the error - see below

Also I cannot find the transfer folder for wax - that I need to load existing files into ax


Traceback (most recent call last):
  File "/home/mike/wax-config/waxconfig.py", line 18, in <module>
    from topbox import top_box
  File "/home/mike/wax-config/topbox.py", line 6, in <module>
    from notebook import notebook
  File "/home/mike/wax-config/notebook.py", line 71, in <module>
    notebook = Notebook()
  File "/home/mike/wax-config/notebook.py", line 32, in __init__
    page = importlib.import_module(module_name)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/home/mike/wax-config/info.py", line 144, in <module>
    page_widget = InfoBox()
  File "/home/mike/wax-config/info.py", line 43, in __init__
    self.list_by_props()
  File "/home/mike/wax-config/info.py", line 100, in list_by_props
    with shelve.open(LONG, 'r') as recording_shelf:
  File "/usr/lib/python3.10/shelve.py", line 243, in open
    return DbfilenameShelf(filename, flag, protocol, writeback)
  File "/usr/lib/python3.10/shelve.py", line 227, in __init__
    Shelf.__init__(self, dbm.open(filename, flag), protocol, writeback)
  File "/usr/lib/python3.10/dbm/__init__.py", line 78, in open
    result = whichdb(file) if 'n' not in flag else None
  File "/usr/lib/python3.10/dbm/__init__.py", line 113, in whichdb
    f = io.open(filename + ".pag", "rb")
TypeError: unsupported operand type(s) for +: 'PosixPath' and 'str'

**Comment by jeffbarish** [comment]
The top folder for wax is ~/wax. In it, there is a hidden directory for
configuration-related stuff called .config. One of the folders in .config
is transfer.

A case could be made that transfer should be in a more obvious place. The
two obvious possibilities are ~/wax or ~/wax/recordings. The first option
makes it too prominent, in my opinion. The transfer folder would be as
prominent as the recordings folder, which contains all the sound files *and*
metadata for the collection. However, the second possibility makes it only
as prominent as the sound directory, which makes sense: The sound directory
is the home of sound files that have been incorporated in the wax database
whereas the transfer directory is the home of sound files that are going to
be incorporated. Moreover, many Wax users will mount a large-capacity disk
drive on recordings, so putting transfer in recordings would confine *all*
sound files to the bulk storage device. Users likely would back up the
recordings drive separately from the root drive, so it also would be
sensible to make transfer sound files part of this backup rather than the
root backup. I like this idea. Expect an update later today.

As for the traceback, I suspect a version problem. To begin with, you are
running python 3.10 whereas I run 3.12.3. Version 3.12 was the default for
the Windows OS I installed. Are you running Windows 11? Are you running WSL
2? You can run uname -a to see which version of WSL you are running. The
long string that I get includes "WSL2". What version of Ubuntu are you
running in WSL? (My WSL came with version 24.04.1 LTS). Run lsb_release -a
to get the Ubuntu version.

On Sat, Dec 21, 2024 at 12:42 AM mikel963 ***@***.***> wrote:

> This looks interesting and I am trying to install on WSL but having a few
> issues
> Wax runs but wax-config does not - with the error - see below
>
> Also I cannot find the transfer folder for wax - that I need to load
> existing files into ax
>
> Traceback (most recent call last):
> File "/home/mike/wax-config/waxconfig.py", line 18, in
> from topbox import top_box
> File "/home/mike/wax-config/topbox.py", line 6, in
> from notebook import notebook
> File "/home/mike/wax-config/notebook.py", line 71, in
> notebook = Notebook()
> File "/home/mike/wax-config/notebook.py", line 32, in *init*
> page = importlib.import_module(module_name)
> File "/usr/lib/python3.10/importlib/*init*.py", line 126, in import_module
> return _bootstrap._gcd_import(name[level:], package, level)
> File "/home/mike/wax-config/info.py", line 144, in
> page_widget = InfoBox()
> File "/home/mike/wax-config/info.py", line 43, in *init*
> self.list_by_props()
> File "/home/mike/wax-config/info.py", line 100, in list_by_props
> with shelve.open(LONG, 'r') as recording_shelf:
> File "/usr/lib/python3.10/shelve.py", line 243, in open
> return DbfilenameShelf(filename, flag, protocol, writeback)
> File "/usr/lib/python3.10/shelve.py", line 227, in *init*
> Shelf.*init*(self, dbm.open(filename, flag), protocol, writeback)
> File "/usr/lib/python3.10/dbm/*init*.py", line 78, in open
> result = whichdb(file) if 'n' not in flag else None
> File "/usr/lib/python3.10/dbm/*init*.py", line 113, in whichdb
> f = io.open(filename + ".pag", "rb")
> TypeError: unsupported operand type(s) for +: 'PosixPath' and 'str'
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/jeffbarish/wax/discussions/3#discussioncomment-11636094>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AY47CTLFKOI54ZBQ5NJX3CL2GULU7AVCNFSM6AAAAABT3HNBJ2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCNRTGYYDSNA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 
Jeffrey Barish

**Comment by mikel963** [comment]
Hi jeff,
many thanks - that helps config runs but wax does not 
Ubuntu 24.04.1 LTS
Python 3.12.3 

I got lots of missing modules - which I started to resolve then realised after a reinstall of Ubuntu - I probably had do a full reinstall of wax>

Anyway after a few more attempts got it to work and uploaded an album - got a few errors with missing directories in the recordings folder - so I added those (sound, documents, images) and then I added an album - hooray !

Many thanks for the help

Will need an easier way to add loads of albums but will persevere for now

**Comment by jeffbarish** [comment]
Glad to hear that you got Wax to run, but you got ahead of me. The version of Wax in the repository now has the transfer folder in recordings. You do not need to worry about it until you upgrade, but beware that you will then have to move your transfer folder.
